src/index.ts

```diff
@@ -1,63 +1,133 @@
-import Capture from './Capture'
-import Auth from './Auth'
-import License from './License'
-import KeyService from './Key'
-import Utils from './utils'
-import { Photo } from '@capacitor/camera'
-
-
-export default class TikiClient{
-  private keyService = new KeyService()
-  private address: string = '';
-  public capture = new Capture()
-  public auth = new Auth(this.keyService)
-  public license = new License()
-
-  public async initialize(providerId: string, pubKey: string, userId: string){
-
-    await this.keyService.clear()
-
-    const keys = await this.keyService.get()
-
-    if(keys.find((key)=> key.value.name === `${providerId}.${userId}`)) return 
-
-    const address = await this.auth.registerAddress(providerId, pubKey, userId)
-
-    this.address = address
+import Capture from "./Capture";
+import Auth from "./Auth";
+import License from "./License";
+import KeyService from "./Key";
+import Utils from "./utils";
+import { Photo } from "@capacitor/camera";
+import { SavedKey } from "./Key/types";
+import type { PostGuardRequest, RspGuard, PostLicenseRequest} from "./License/types";
+
+export default class TikiClient {
+  private static keyService = new KeyService();
+
+  public static capture = new Capture();
+  public static auth = new Auth(TikiClient.keyService);
+  public static license = new License();
+
+  private constructor() {}
+
+  /**
+   * Initialize the TikiClient and register the device's address.
+   * @param {string} providerId - the provider ID of the associated provider account.
+   * @param {string} pubKey - the public key of the created provider.
+   * @param {string} userId - the ID to be registered to identify the user.
+   */
+  public static async initialize(
+    providerId: string,
+    pubKey: string,
+    userId: string
+  ): Promise<void> {
+    await TikiClient.keyService.clear();
+
+    const keys = await TikiClient.keyService.get();
+
+    if (keys.find((key) => key.value.name === `${providerId}.${userId}`))
+      throw new Error(
+        "TThe address is already registered for these provider and user IDs."
+      );
+
+    await TikiClient.auth.registerAddress(providerId, pubKey, userId);
+  }
 
+  /**
+   * Capture a picture and send it to Tiki.
+   * Uses the capture module to take the photo and publish it to Tiki.
+   * Also utilizes the license module to verify if the provided license is valid.
+   * @param {string} providerId - the provider ID of the associated provider account.
+   * @param {string} userId - The user ID to link the receipt to their information.
+   * @param {PostGuardRequest} licenseReq - the License pointer record and use cases to verify if the license is valid.
+   * @param {string} requestId - a UUID string to identify the location of the pictures that are sent.
+   */
+  public static async scan(
```

Comment:
The only parameter of the scan is the requestId.
providerId and userId needs to be set as class public parameters.

The PostGuardRequest is hardcoded:
ptr: userId
uses: [new Use(usecase: new Usecase(value: 'ATRIBUTION'))], destinations: ["*"]

Reply:
https://github.com/tiki/publish-client-capacitor/issues/44